mount/volumemountutils_windows.go

```diff
@@ -0,0 +1,76 @@
+// +build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+// Simple wrappers around SetVolumeMountPoint and DeleteVolumeMountPoint
+
+import (
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// Mount volumePath (in format '\\?\Volume{GUID}' at targetPath.
+// https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setvolumemountpointw
+func setVolumeMountPoint(targetPath string, volumePath string) error {
+	if !strings.HasPrefix(volumePath, "\\\\?\\Volume{") {
+		return errors.Errorf("unable to mount non-volume path %s", volumePath)
+	}
+
+	// Both must end in a backslash
+	slashedTarget := filepath.Clean(targetPath) + string(filepath.Separator)
+	slashedVolume := volumePath + string(filepath.Separator)
+
+	targetP, err := syscall.UTF16PtrFromString(slashedTarget)
```

Comment:
Per https://github.com/microsoft/hcsshim/pull/852#discussion_r461233781, use `windows.UTF16PtrFromString`. In fact, update this whole file to be the same as the one I added to hcsshim. Or put together a PR for these and related functions to address https://github.com/microsoft/go-winio/issues/176, and then consume them from there.